web_programming/current_weather.py

```diff
@@ -1,30 +1,64 @@
 import requests
 
-APPID = ""  # <-- Put your OpenWeatherMap appid here!
-URL_BASE = "https://api.openweathermap.org/data/2.5/"
+# Replace these API keys with your respective API keys
+OPENWEATHERMAP_API_KEY = ""  # <-- Put your OpenWeatherMap API key here!
+WEATHERSTACK_API_KEY = ""  # <-- Put your Weatherstack API key here!
 
+# Define the URL for OpenWeatherMap API with placeholders
+OPENWEATHERMAP_URL_BASE = "https://api.openweathermap.org/data/2.5/weather"
+WEATHERSTACK_URL_BASE = "http://api.weatherstack.com/current"
 
-def current_weather(q: str = "Chicago", appid: str = APPID) -> dict:
-    """https://openweathermap.org/api"""
-    return requests.get(URL_BASE + "weather", params=locals()).json()
-
-
-def weather_forecast(q: str = "Kolkata, India", appid: str = APPID) -> dict:
-    """https://openweathermap.org/forecast5"""
-    return requests.get(URL_BASE + "forecast", params=locals()).json()
 
+def current_weather(location: str, api_provider: str, api_key: str) -> dict:
+    if api_provider == "openweathermap":
```

Comment:
Super cool but let's allow side-by-side comparisons of forecasts from both services.

We could drive this by the presence / absence of API keys.
`def current_weather(location: str) -> tuple[dict]:`
* if not OPENWEATHERMAP_API_KEY not WEATHERSTACK_API_KEY then raise `ValueError`
* if both OPENWEATHERMAP_API_KEY and WEATHERSTACK_API_KEY then add both dicts to the tuple
* if only one API key is defined then add only one dict to the tuple.